codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -0,0 +1,174 @@
+/**
+ * File: array_hash_map.c
+ * Created Time: 2023-03-18
+ * Author: Guanngxu (446678850@qq.com)
+ */
+
+#include "../include/include.h"
+
+/* 哈希表初始化函数 */
+array_hash_map_t *new_array_hash_map() {
+    array_hash_map_t *array_hash_map = malloc(sizeof(array_hash_map_t));
+    return array_hash_map;
+}
+
+/* 哈希函数 */
+int hash_func(int key) {
+    int index = key % HASH_MAP_DEFAULT_SIZE;
+    return index;
+}
+
+/* 查询操作 */
+const char *get(const array_hash_map_t *d, const int key) {
+    int index = hash_func(key);
+    const entry_t *pair = d->buckets + index;
```

Comment:
It is best to be consistent with cpp, `const entry_t *pair = d->buckets[index];` easy to understand